src/OpenSpaceToolkit/Astrodynamics/Trajectory/LocalOrbitalFrameTransformProvider.cpp

```diff
@@ -119,18 +116,20 @@ String LocalOrbitalFrameTransformProvider::StringFromType(const LocalOrbitalFram
     return String::Empty();
 }
 
-Transform LocalOrbitalFrameTransformProvider::generateTransform(
-    const LocalOrbitalFrameTransformProvider::Type& aType,
-    const Instant& anInstant,
-    const Vector3d& aPosition,
-    const Vector3d& aVelocity
+Transform LocalOrbitalFrameTransformProvider::GenerateTransform(
+    const LocalOrbitalFrameTransformProvider::Type& aType, const State& aState
 )
 {
+    const Instant& instant = aState.accessInstant();
+    const Vector3d positionCoordinates = aState.getPosition().accessCoordinates();
+    const Vector3d velocityCoordinates = aState.getVelocity().accessCoordinates();
```

Comment:
These are sure to be given in GCRF right?

Reply:
They're not, they will be in whatever parent frame the user has provided, which can be GCRF. Typically these are provided by the localFrameFactory and in the correct frame. So I think we're covered.